### PR TITLE
[Docs] [Enterprise Search] ML inference pipeline documentation updates

### PR DESCRIPTION
**Pull request by demjened**
Updates to ML inference pipeline documentation, as per UX changes in 8.12. See explanations and context in inline comments.

### CODE REVIEW

File: docs/reference/ingest/search-inference-processing.asciidoc [hunk 1]
```diff
@@ -123,50 +126,10 @@ These pipelines can also be viewed, edited, and deleted in Kibana via *Stack Man
 You may also use the <<ingest-apis,Ingest pipeline APIs>>.
 If you delete any of these pipelines outside of the *Content* UI in Kibana, make sure to edit the ML inference pipelines that reference them.
 
-[discrete#ingest-pipeline-search-inference-update-mapping]
```
Comment (demjened) [inline docs/reference/ingest/search-inference-processing.asciidoc:126]: The "Mappings" step has been removed from the pipeline creation workflow in 8.12.

File: docs/reference/ingest/search-inference-processing.asciidoc [hunk 2]
```diff
@@ -102,10 +102,13 @@ Here, you'll be able to:
 1. Choose a name for your pipeline.
   - This name will need to be unique across the whole deployment.
   If you want this pipeline to be index-specific, we recommend including the name of your index in the pipeline name.
+  - If you do not set the pipeline name, a default unique name will be provided upon selecting an ML trained model.
```
Comment (demjened) [inline docs/reference/ingest/search-inference-processing.asciidoc:105]: If the user doesn't touch the pipeline name field, then upon selecting a model, the name will be automatically generated using the `ml-inference-<index name>-<normalized model name>` pattern. Normalization of the model name means every non-alphanumeric symbol other than "-" is converted to underscore.

But if the pipeline name is edited, selecting a model does not change it.

Screenshot 2023-12-28 at 11 54 34

File: docs/reference/ingest/search-inference-processing.asciidoc [hunk 3]
```diff
@@ -102,10 +102,13 @@ Here, you'll be able to:
...
 2. Select the ML trained model you want to use.
+  - The model must be deployed before you can select it.
+  To begin deployment of a model, click the *Deploy* button.
```
Comment (demjened) [inline docs/reference/ingest/search-inference-processing.asciidoc:108]: Some curated models (E5 and ELSER at this time) are pinned to the top of the list even if they are not deployed to the cluster. The user can select either of these, however the pipeline creation cannot continue unless the model is deploying/deployed/started (and so the Continue button is disabled).

The deployment can be started from this screen using the Deploy button, which begins the deployment of the selected model and enables the Continue button right away.


Screenshot 2023-12-28 at 11 54 42

(This doesn't apply to a 3rd party model, which is already in the deployed state after the user uploads it to the cluster.)

File: docs/reference/ingest/search-inference-processing.asciidoc [hunk 4]
```diff
@@ -123,50 +126,10 @@ These pipelines can also be viewed, edited, and deleted in Kibana via *Stack Man
...
-[TIP]
-====
-You can check the shape of the generated output before indexing any documents while creating the ML inference pipeline under the *Test* tab.
-Simply provide a sample document, click *Simulate*, and look for the `ml.inference` object in the results.
-====
-
 [discrete#ingest-pipeline-search-inference-test-inference-pipeline]
 ==== Test your ML inference pipeline
+You can check the expected structure of the inference output before indexing any documents while creating the ML inference pipeline under the *Test* tab.
+Simply provide a sample document, click *Simulate*, and look for the `ml.inference` object in the results.
```
Comment (szabosteve) [inline docs/reference/ingest/search-inference-processing.asciidoc:132]:
```suggestion

You can check the expected structure of the inference output before indexing any documents while creating the {ml} inference pipeline under the *Test* tab.
Provide a sample document, click **Simulate**, and look for the `ml.inference` object in the results.
```

**Review by szabosteve — APPROVED** [review]
I left one tiny suggestion. Please take or leave it. Otherwise, LGTM! Thank you for adjusting the docs.

File: docs/reference/ingest/search-inference-processing.asciidoc (hunk 2 above)
Comment (kderusso) [inline docs/reference/ingest/search-inference-processing.asciidoc:105]: > an ML trained model.

I'm not sure about grammar here whether this should be `a` or `an`? You could consider changing it to `a trained model` to remove the grammar ambiguity.

File: docs/reference/ingest/search-inference-processing.asciidoc [hunk 5]
```diff
@@ -123,50 +126,10 @@ These pipelines can also be viewed, edited, and deleted in Kibana via *Stack Man
...
-
-[TIP]
-====
-You can check the shape of the generated output before indexing any documents while creating the ML inference pipeline under the *Test* tab.
-Simply provide a sample document, click *Simulate*, and look for the `ml.inference` object in the results.
-====
-
 [discrete#ingest-pipeline-search-inference-test-inference-pipeline]
 ==== Test your ML inference pipeline
+You can check the expected structure of the inference output before indexing any documents while creating the ML inference pipeline under the *Test* tab.
```
Comment (kderusso) [inline docs/reference/ingest/search-inference-processing.asciidoc:131]:
```suggestion
You can verify the expected structure of the inference output before indexing any documents while creating the ML inference pipeline under the *Test* tab.
```

**Review by kderusso — APPROVED** [review]
LGTM, two nitpicky suggestions.

**Pull request merged**